lib/ecto/adapters/sql.ex

```diff
@@ -245,6 +245,73 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
+  @doc """
+  Executes an EXPLAIN statement or similar for the given query according to its kind and the
+  adapter in the given repository.
+
+  ## Examples
+
+      # Postgres
+      iex> Ecto.Adapters.SQL.explain(:all, Repo, Post)
+      "Seq Scan on posts p0  (cost=0.00..12.12 rows=1 width=443)"
+
+      # MySQL
+      iex> Ecto.Adapters.SQL.explain(:all, from(p in Post, where: p.title == "title")) |> IO.puts()
+      +----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+      | id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra       |
+      +----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+      |  1 | SIMPLE      | p0    | NULL       | ALL  | NULL          | NULL | NULL    | NULL |    1 |    100.0 | Using where |
+      +----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------------+
+
+  It's safe to execute it for updates and deletes, no data change will be commited:
+
+      iex> Ecto.Adapters.SQL.explain(:update_all, Repo, from(p in Post, update: [set: [title: "new title"]]))
+      "Update on posts p0  (cost=0.00..11.70 rows=170 width=449)\n  ->  Seq Scan on posts p0  (cost=0.00..11.70 rows=170 width=449)"
+
+  Valid `params` will be mapped directly to the EXPLAIN statement's params for the adapter in use,
+  so please note each adapter may have specific params:
```

Comment:
I think we can have a section in the docs listing the options for the builtin adapters. And then we mention that if you are not using a builtin adapter, you should consult their docs. WDYT?

Reply:
Something like that? https://github.com/elixir-ecto/ecto_sql/pull/231/commits/1efd817890e6e864ea5bcd1762dcefe24b906273

<img width="789" alt="Screen Shot 2020-06-06 at 12 44 06 PM" src="https://user-images.githubusercontent.com/36407/83949768-8f392580-a7f3-11ea-809e-d64010c8a2f7.png">

The table doesn't render right 🤔 
Can you take a look?